scripts/ci/pairing/pkg/cmd/pair.go

```
@@ -0,0 +1,176 @@
+
+	// list the open pull requests
+	opt := &github.PullRequestListOptions{
+		State: "open",
+	}
+
+	prs, _, err := client.PullRequests.List(context.Background(), owner, repo, opt)
+	if err != nil {
+		return nil, fmt.Errorf("could not list pull requests: %w", err)
+	}
+
+	return prs, nil
+}
+
+// shouldPair determines whether the given remote and branch name should be paired
+// based on existing open pull requests in the given organization and repository.
+func shouldPair(orgForPairing, repoForPairing, currentRemoteName, currentBranchName string) (bool, error) {
+	pullRequests, err := listOpenPRs(orgForPairing, repoForPairing)
+	if err != nil {
+		return false, err
+	}
+
+	for _, pull := range pullRequests {
+		if pull.GetHead().GetRef() == currentBranchName && pull.GetUser().GetLogin() == currentRemoteName {
+			return true, nil
+		}
+	}
+
+	return false, nil
+}
+
+// getCurrentPRInfo gets the current info of the PR that triggered the pairing
+// the pairing can be triggered by CI job or GH action
+func getCurrentPRInfo() (*PullRequestMetadata, error) {
+	pr := &PullRequestMetadata{}
+	jobSpecEnvVarData := os.Getenv("JOB_SPEC")
+
+	// running CI job
+	if jobSpecEnvVarData != "" {
+		log.Println("running in CI job")
+		jobSpec := &JobSpec{}
+		if err := json.Unmarshal([]byte(jobSpecEnvVarData), jobSpec); err != nil {
+			return pr, fmt.Errorf("error when parsing openshift job spec data: %w", err)
+		}
+		if len(jobSpec.Refs.Pulls) == 1 {
+			pull := jobSpec.Refs.Pulls[0]
+			pr = &PullRequestMetadata{
+				RemoteName: pull.Author,
+				Number:     strconv.Itoa(pull.Number),
+				BranchName: pull.HeadRef,
+			}
+		} else {
+			fmt.Println("No pull request data found.")
+		}
+		// running GH action
+	} else if os.Getenv("GITHUB_ACTIONS") != "" {
+		log.Println("running in GH action")
+		pr = &PullRequestMetadata{
+			RemoteName: os.Getenv("AUTHOR"),
+			BranchName: os.Getenv("GITHUB_HEAD_REF"),
+		}
+	}
+
```

if neither OpenShift-CI nor GH actions are recognized, then we should return an error

---

I forgot the detail that when neither OpenShift-CI nor GH actions, we need to clone the parent repo